pysheaf/pysheaf.py

```diff
@@ -827,6 +835,278 @@ def serializeAssignment(self,assignment):
                 x0[idxarray[cell.support]:idxarray[cell.support]+self.cells[cell.support].stalkDim]=cell.value
 
         return x0,bounds
+                
+    
+    def optimize_SLSQP(self, assignment, tol=1e-5):
+        """
+        Compute the nearest global section to a given assignment using 
+        scipy.optimize.minimize. When there are constraints specified, 
+        scipy.optimize.minimize defaults to using SLSQP.
+        Based on:
+            Kraft, D. A software package for sequential quadratic programming. 1988. 
+            Tech. Rep. DFVLR-FB 88-28, DLR German Aerospace Center, 
+            Institute for Flight Mechanics, Koln, Germany.
+        """
+    
+        initial_guess, bounds = self.serializeAssignment(assignment)
+        res=scipy.optimize.minimize( fun = lambda sec: self.assignmentMetric(assignment,self.deserializeAssignment(sec)),
+                                    x0 = initial_guess,
+                                    method = 'SLSQP', 
+                                    bounds = bounds,
+                                    constraints = ({'type' : 'eq',
+                                                    'fun' : lambda asg: self.consistencyRadius(self.deserializeAssignment(asg))}), 
+                                    tol = tol, 
+                                    options = {'maxiter' : int(1000)})
+        globalsection = self.deserializeAssignment(res.x)
+        #print res.success
+        #print res.message
+        return globalsection
+    
+    
+    
+    
+    def ga_optimization_function(self,  space_des_2_opt, assignment,individual):
+        """Write the function for the genetic algorithm to optimize similar to fun for scipy.optimize.minimize"""
+        
+        #Assign a high cost to eliminate individuals with nans from the population
+        if np.any(np.isnan(individual)):
+            cost = 1e100
+            cost = -cost
+            
+        else:
+            #write a new assignment from the individual so that one can use maximal extend.
+            new_assignment = []
+            start_index = 0
+            for i in range(len(space_des_2_opt)):
+                new_assignment.append(SectionCell(support=space_des_2_opt[i][0], value=individual[(start_index):(start_index+space_des_2_opt[i][1])]))
+                start_index += space_des_2_opt[i][1]
+        
+            new_assignment = Section(new_assignment)
+        
+            #start of optimization function
+            new_assignment = self.maximalExtend(new_assignment,multiassign=False,tol=1e-5)
+        
+            cost = self.assignmentMetric(assignment, new_assignment)
+            #Sum to formulate cost (NOTE: GA maximizes instead of minimizes so we need a negative sign)
+            cost = -cost
+        return (float(cost),)
+        
+
+    
+    def optimize_GA(self, assignment, tol=1e-5, initial_pop_size=100, mutation_rate = .3, num_generations=100 ,num_ele_Hallfame=1):
```

Comment:
Should be described in the documentation, near where fuseAssignment is discussed.  Additionally, this should have the same (essentially) I/O spec as fuseAssignment, with the exception that it can take additional optional arguments.